### ISSUE TITLE
How to mock System with 1.16

### ISSUE THREAD

**Issue by ascassel** [jmockit/jmockit1#155]
I used to mock java.System to ensure that the tested code uses System.exit() with the expected return code - but so far I only get java.lang.IllegalArgumentException: Matching real methods not found for the following mocks:
test.Test$4#exit(int)
	at test.Test$4.<init>(Test.java:178)
   ......
new MockUp&lt;System&gt;()
 {
    @Mock
    public void exit( int code )
    {
       //verification code here
    }
};   
....

Is this no longer possible? Note that I use a IBM JAVA 7

**Comment by marcdonovan** [comment]
Don't mock java internals. It only leads to misery. Ditto for using IBM java. If you are not on a mainframe, there is no excuse.

**Comment by rliesenfeld** [comment]
I am unable to see any problem. The following test works as expected:

```java
    @Test
    public void mockSystemExit() {
        new MockUp<System>() {
            @Mock
            void exit(int i) {}
        };

        System.exit(1);
    }
```

**Comment by ascassel** [comment]
Rogiero, thx for your answer.
You are right, it works in an Oracle JVM - but fails using the IBM one.
And yes, I need to be running on ZOS (thx Marc for the helpful comment... 8-) )
What frustrates more is that it worked in the 1.9 version.. 
Regards,
Achim

**Comment by rliesenfeld** [comment]
I discovered why this isn't working anymore. It is described in the release notes for version 1.11: http://jmockit.org/changes.html#1.11

On the IBM JRE, java.lang.System gets explicitly excluded from mocking with a MockUp. This was done after seeing the test execution thread hanging up, when said class was mocked. It appears to be a bug in the IBM JVM; so, for safety, I decided to exclude java.lang.System entirely.

This isn't the only problem with the IBM JRE. On a developer machine which likely can use an Oracle JDK, I would recommed to use that for test execution. But if you have a ZOS build machine which can only use the IBM JRE, then there is no choice but to avoid mocking the System class.
